workout-logger/scripts/test_gemini_api.py

```
@@ -0,0 +1,306 @@
+                    current_model = fallback
+                    continue
+
+            if e.code in (429, 500, 502, 503, 504):
+                retry_sec = extract_retry_delay(body) or (0.5 * (2**attempt))
+                print(f"  [HTTP {e.code}] Rate limit/server error detected. Google retryDelay: {retry_sec:.2f}s (Attempt {attempt+1}/{max_attempts})")
+                if attempt < max_attempts - 1:
+                    print(f"  --> Waiting {retry_sec:.2f}s before retry...")
+                    time.sleep(retry_sec)
+                    continue
+            print(f"\n[!] HTTP {e.code} Error Body:\n{body}")
+            raise e
```

Ruff (0.16.1)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)